src/benchmarklib/tpch/tpch_benchmark_item_runner.cpp

```diff
@@ -71,11 +71,13 @@ void TPCHBenchmarkItemRunner::on_tables_loaded() {
   if (_clustering_configuration == ClusteringConfiguration::Pruning) {
     Assert(!first_chunk->individually_sorted_by().empty(), "Sorting information was lost");
   }
-
-  if (_config->indexes) {
+  if (_config->chunk_indexes) {
     const auto indexed_column_ids = std::vector<ColumnID>{ColumnID{0}};
     Assert(!first_chunk->get_indexes(indexed_column_ids).empty(), "Index was lost");
   }
+  if (_config->table_indexes) {
+    Assert(!orders_table->get_table_indexes().empty(), "Index was lost");
```

Comment:
It should not happen, but with faulty code modifications, this might happen. I guess this assertion shall help to observe when incorrect code was added. I think this should rather be checked in a unit test, e.g., `tpch_db_generator_test` (see `TPCHTableGeneratorTest, GenerateAndStore`).